In [13]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt



## Import Data
- We are importing data from a specified location  

In [14]:
data = pd.read_csv("dataset/minor_dataset.csv")

In [15]:
data.head()

itching  skin_rash  nodal_skin_eruptions  continuous_sneezing  shivering  \
0        1          1                     1                    0          0   
1        0          1                     1                    0          0   
2        1          0                     1                    0          0   
3        1          1                     0                    0          0   
4        1          1                     1                    0          0   

   chills  joint_pain  stomach_pain  acidity  ulcers_on_tongue  \
0       0           0             0        0                 0   
1       0           0             0        0                 0   
2       0           0             0        0                 0   
3       0           0             0        0                 0   
4       0           0             0        0                 0   

   muscle_wasting  vomiting  burning_micturition  spotting_ urination  \
0               0         0                    0                    0   
1               0         0                    0                    0   
2               0         0                    0                    0   
3               0         0                    0                    0   
4               0         0                    0                    0   

   fatigue  weight_gain  anxiety  cold_hands_and_feets  mood_swings  \
0        0            0        0                     0            0   
1        0            0        0                     0            0   
2        0            0        0                     0            0   
3        0            0        0                     0            0   
4        0            0        0                     0            0   

   weight_loss  restlessness  lethargy  patches_in_throat  \
0            0             0         0                  0   
1            0             0         0                  0   
2            0             0         0                  0   
3            0             0         0                  0   
4            0             0         0                  0   

   irregular_sugar_level  cough  high_fever  sunken_eyes  breathlessness  \
0                      0      0           0            0               0   
1                      0      0           0            0               0   
2                      0      0           0            0               0   
3                      0      0           0            0               0   
4                      0      0           0            0               0   

   sweating  dehydration  indigestion  headache  yellowish_skin  dark_urine  \
0         0            0            0         0               0           0   
1         0            0            0         0               0           0   
2         0            0            0         0               0           0   
3         0            0            0         0               0           0   
4         0            0            0         0               0           0   

   nausea  loss_of_appetite  pain_behind_the_eyes  back_pain  constipation  \
0       0                 0                     0          0             0   
1       0                 0                     0          0             0   
2       0                 0                     0          0             0   
3       0                 0                     0          0             0   
4       0                 0                     0          0             0   

   abdominal_pain  diarrhoea  mild_fever  yellow_urine  yellowing_of_eyes  \
0               0          0           0             0                  0   
1               0          0           0             0                  0   
2               0          0           0             0                  0   
3               0          0           0             0                  0   
4               0          0           0             0                  0   

   acute_liver_failure  fluid_over

## Checking the Dataset
- In the upcoming cells we will be checking our dataset for null values 
- We will also perform exploratory data analysis(EDA) on our dataset

In [16]:
data.shape

(4961, 133)

- We have 4961 rows and 133 columns in our dataset 

## Checkingfor null values

- Our dataset cotains 0 null values 

In [19]:
null_values = data.isnull().sum()
print(null_values)

itching                           0
skin_rash                         0
nodal_skin_eruptions              0
continuous_sneezing               0
shivering                         0
chills                            0
joint_pain                        0
stomach_pain                      0
acidity                           0
ulcers_on_tongue                  0
muscle_wasting                    0
vomiting                          0
burning_micturition               0
spotting_ urination               0
fatigue                           0
weight_gain                       0
anxiety                           0
cold_hands_and_feets              0
mood_swings                       0
weight_loss                       0
restlessness                      0
lethargy                          0
patches_in_throat                 0
irregular_sugar_level             0
cough                             0
high_fever                        0
sunken_eyes                       0
breathlessness              

- checking for uniques values in each column

In [20]:
data.nunique()

itching                            2
skin_rash                          2
nodal_skin_eruptions               2
continuous_sneezing                2
shivering                          2
chills                             2
joint_pain                         2
stomach_pain                       2
acidity                            2
ulcers_on_tongue                   2
muscle_wasting                     2
vomiting                           2
burning_micturition                2
spotting_ urination                2
fatigue                            2
weight_gain                        2
anxiety                            2
cold_hands_and_feets               2
mood_swings                        2
weight_loss                        2
restlessness                       2
lethargy                           2
patches_in_throat                  2
irregular_sugar_level              2
cough                              2
high_fever                         2
sunken_eyes                        2
b

### Now we have looked at our dataset and came to following conclusions :-

- Our Dataset contains 4961 rows and 133 columns
- Our dataset contains Zero(0) Null values
- Our dataset is one_hot_encoded
 


## Distribution of outcome variable

- Now we will check for distribution of each label in our outcome variable

In [21]:
data.head()

itching  skin_rash  nodal_skin_eruptions  continuous_sneezing  shivering  \
0        1          1                     1                    0          0   
1        0          1                     1                    0          0   
2        1          0                     1                    0          0   
3        1          1                     0                    0          0   
4        1          1                     1                    0          0   

   chills  joint_pain  stomach_pain  acidity  ulcers_on_tongue  \
0       0           0             0        0                 0   
1       0           0             0        0                 0   
2       0           0             0        0                 0   
3       0           0             0        0                 0   
4       0           0             0        0                 0   

   muscle_wasting  vomiting  burning_micturition  spotting_ urination  \
0               0         0                    0                    0   
1               0         0                    0                    0   
2               0         0                    0                    0   
3               0         0                    0                    0   
4               0         0                    0                    0   

   fatigue  weight_gain  anxiety  cold_hands_and_feets  mood_swings  \
0        0            0        0                     0            0   
1        0            0        0                     0            0   
2        0            0        0                     0            0   
3        0            0        0                     0            0   
4        0            0        0                     0            0   

   weight_loss  restlessness  lethargy  patches_in_throat  \
0            0             0         0                  0   
1            0             0         0                  0   
2            0             0         0                  0   
3            0             0         0                  0   
4            0             0         0                  0   

   irregular_sugar_level  cough  high_fever  sunken_eyes  breathlessness  \
0                      0      0           0            0               0   
1                      0      0           0            0               0   
2                      0      0           0            0               0   
3                      0      0           0            0               0   
4                      0      0           0            0               0   

   sweating  dehydration  indigestion  headache  yellowish_skin  dark_urine  \
0         0            0            0         0               0           0   
1         0            0            0         0               0           0   
2         0            0            0         0               0           0   
3         0            0            0         0               0           0   
4         0            0            0         0               0           0   

   nausea  loss_of_appetite  pain_behind_the_eyes  back_pain  constipation  \
0       0                 0                     0          0             0   
1       0                 0                     0          0             0   
2       0                 0                     0          0             0   
3       0                 0                     0          0             0   
4       0                 0                     0          0             0   

   abdominal_pain  diarrhoea  mild_fever  yellow_urine  yellowing_of_eyes  \
0               0          0           0             0                  0   
1               0          0           0             0                  0   
2               0          0           0             0                  0   
3               0          0           0             0                  0   
4               0          0           0             0                  0   

   acute_liver_failure  fluid_over

In [22]:
data['prognosis'].value_counts(normalize=True)

prognosis
Fungal Infection                 0.02439
Allergy                          0.02439
GERD                             0.02439
Chronic Cholestasis              0.02439
Drug Reaction                    0.02439
Peptic Ulcer Disease             0.02439
AIDS                             0.02439
Diabetes                         0.02439
Gastroenteritis                  0.02439
Bronchial Asthma                 0.02439
Hypertension                     0.02439
Migraine                         0.02439
Cervical Spondylosis             0.02439
Paralysis (brain hemorrhage)     0.02439
Jaundice                         0.02439
Malaria                          0.02439
Chickenpox                       0.02439
Dengue                           0.02439
Typhoid                          0.02439
Hepatitis A                      0.02439
Hepatitis B                      0.02439
Hepatitis C                      0.02439
Hepatitis D                      0.02439
Hepatitis E                      0.02439
Alcoho

- we see that our dataset's outcome variable is normaly distributed

In [23]:
data.describe()

itching    skin_rash  nodal_skin_eruptions  continuous_sneezing  \
count  4961.000000  4961.000000           4961.000000          4961.000000   
mean      0.137875     0.159847              0.021971             0.045152   
std       0.344804     0.366501              0.146605             0.207659   
min       0.000000     0.000000              0.000000             0.000000   
25%       0.000000     0.000000              0.000000             0.000000   
50%       0.000000     0.000000              0.000000             0.000000   
75%       0.000000     0.000000              0.000000             0.000000   
max       1.000000     1.000000              1.000000             1.000000   

         shivering       chills   joint_pain  stomach_pain      acidity  \
count  4961.000000  4961.000000  4961.000000   4961.000000  4961.000000   
mean      0.021971     0.162266     0.139085      0.045152     0.045152   
std       0.146605     0.368732     0.346070      0.207659     0.207659   
min       0.000000     0.000000     0.000000      0.000000     0.000000   
25%       0.000000     0.000000     0.000000      0.000000     0.000000   
50%       0.000000     0.000000     0.000000      0.000000     0.000000   
75%       0.000000     0.000000     0.000000      0.000000     0.000000   
max       1.000000     1.000000     1.000000      1.000000     1.000000   

       ulcers_on_tongue  muscle_wasting     vomiting  burning_micturition  \
count       4961.000000     4961.000000  4961.000000          4961.000000   
mean           0.021971        0.021971     0.389236             0.043943   
std            0.146605        0.146605     0.487626             0.204988   
min            0.000000        0.000000     0.000000             0.000000   
25%            0.000000        0.000000     0.000000             0.000000   
50%            0.000000        0.000000     0.000000             0.000000   
75%            0.000000        0.000000     1.000000             0.000000   
max            1.000000        1.000000     1.000000             1.000000   

       spotting_ urination      fatigue  weight_gain      anxiety  \
count          4961.000000  4961.000000  4961.000000  4961.000000   
mean              0.021971     0.392864     0.023181     0.023181   
std               0.146605     0.488436     0.150493     0.150493   
min               0.000000     0.000000     0.000000     0.000000   
25%               0.000000     0.000000     0.000000     0.000000   
50%               0.000000     0.000000     0.000000     0.000000   
75%               0.000000     1.000000     0.000000     0.000000   
max               1.000000     1.000000     1.000000     1.000000   

       cold_hands_and_feets  mood_swings  weight_loss  restlessness  \
count           4961.000000  4961.000000  4961.000000   4961.000000   
mean               0.023181     0.046362     0.092723      0.046362   
std                0.150493     0.210288     0.290073      0.210288   
min                0.000000     0.000000     0.000000      0.000000   
25%                0.000000     0.000000     0.000000      0.000000   
50%                0.000000     0.000000     0.000000      0.000000   
75%                0.000000     0.000000     0.000000      0.000000   
max                1.000000     1.000000     1.000000      1.000000   

          lethargy  patches_in_throat  irregular_sugar_level        cough  \
count  4961.000000        4961.000000            4961.000000  4961.000000   
mean      0.092723           0.021971               0.023181     0.114695   
std       0.290073           0.146605               0.150493     0.318685   
min       0.000000           0.000000               0.000000     0.000000   
25%       0.000000           0.000000               0.000000     0.000000   
50%       0.000000           0.000000               0.000000     0.000000   
75%       0.000000           0.000000               0.000000     0.000000   
max       1.000000           1.000000               1.000000

## Label Encoding

 Now there is one thing left to do , to label encode our dataset outcome variable to convert it into integers value